poopcam.ts

```diff
@@ -28,14 +29,25 @@ export abstract class PoopCam {
         }
 
         if (cammer === undefined) {
-            cammer = {userName: userName, requestCount: 0};
+            cammer = {userName: userName, requestCount: 0, date: new Date(0)};
 
         }
-
-        cammer.requestCount++;
+        let date: Date = new Date();
+        let rate_limit = false;
+        if (cammer.date === undefined) {
+            rate_limit = true;
```

Comment:
The name of the variable is a bit confusing. Can you rename it to "rate_limited" and reverse the logic? So if rate_limited == true, then the count won't increment.

Reply:
updated